stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py

```diff
@@ -0,0 +1,190 @@
+        # ----- Create a PackagePreparationLine for every stock move
+        #       in the pickings added to PackagePreparation
+        if values.get('picking_ids', False):
+            package_preparation_lines = self.env[
+                'stock.picking.package.preparation.line'
+                ]._prepare_lines_from_pickings(values['picking_ids'][0][2])
+            if package_preparation_lines:
+                values.update({
+                    'line_ids': [(0, 0, v) for v in package_preparation_lines]
+                })
+        return values
+
+    @api.model
+    def create(self, values):
+        values = self._update_line_ids(values)
+        return super(StockPickingPackagePreparation, self).create(values)
+
+    @api.multi
+    def write(self, values):
+        values = self._update_line_ids(values)
+        return super(StockPickingPackagePreparation, self).write(values)
+
+    @api.multi
+    def action_put_in_pack(self):
+        # ----- Create a stock move and stock picking related with
+        #       StockPickingPackagePreparationLine if this one has
+        #       a product in the values
+        picking_model = self.env['stock.picking']
+        move_model = self.env['stock.move']
+        for package in self:
+            picking_type = package.picking_type_id or \
+                self.env.ref('stock.picking_type_out')
+            moves = []
+            for line in package.line_ids:
+                # ----- If line has 'move_id' this means we don't need to
+                #       recreate picking and move again
+                if line.product_id and not line.move_id:
+                    move_data = line.get_move_data()
+                    move_data.update({
+                        'partner_id': package.partner_id.id,
+                        'location_id':
+                            picking_type.default_location_src_id.id,
+                        'location_dest_id':
+                            picking_type.default_location_dest_id.id,
+                        })
+                    moves.append((line, move_data))
+            if moves:
+                picking_data = {
+                    'move_type': 'direct',
+                    'partner_id': package.partner_id.id,
+                    'company_id': package.company_id.id,
+                    'date': package.date,
+                    'picking_type_id': picking_type.id,
+                    }
+                picking = picking_model.create(picking_data)
+                for line, move_data in moves:
+                    move_data.update({'picking_id': picking.id})
+                    move = move_model.create(move_data)
+                    line.move_id = move.id
+                # ----- Set the picking as "To DO" and try to set it as
+                #       assigned
+                picking.action_confirm()
+                # ----- Show an error if a picking is not confirmed
+                if picking.state != 'confirmed':
```

Comment:
My picking is in state assigned at this step (see in debugger). I think it can be ok, no ?